planner/core/plan_to_pb.go

```diff
@@ -139,6 +140,11 @@ func (p *PhysicalIndexScan) ToPB(ctx sessionctx.Context) (*tipb.Executor, error)
 // SetPBColumnsDefaultValue sets the default values of tipb.ColumnInfos.
 func SetPBColumnsDefaultValue(ctx sessionctx.Context, pbColumns []*tipb.ColumnInfo, columns []*model.ColumnInfo) error {
 	for i, c := range columns {
+		// For virtual columns, we set their default values to NULL so that TiKV will return NULL properly,
+		// They real values will be compute later.
+		if c.IsGenerated() && !c.GeneratedStored {
+			pbColumns[i].DefaultVal = []byte{codec.NilFlag}
```

Comment:
How much additional effort would it cost if we want to remove the virtual generated columns from the schema or columns of `PhysicalTableScan`?

Reply:
I can't understand this question, why do we have to move it?

Reply:
We need a place to save the virtual column's expression and its columnInfo, currently it is in `PhysicalTableScan`.